### ISSUE TITLE
New Site Import - Select Call Centre (part II)

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#468]
In GitLab by @smasih on Sep 12, 2017, 23:52

This ticket will detail the exported files for this new Site/Client and how to import their existing data into SAB.

#### [Effort Log for #305 ](http://gitlab.aavaz.biz/AnswerNet/SABilling/wikis/effort-log-305)

**Primary ticket #305**

**Secondary Tickets:**
* #529 **Human Readable for SCC** 
* #530 **Moneris Payment Gateway (Primary Ticket for Moneris)**
* #531 **Data Import for SCC**
* #532 **CC Details for Moneris Vault**
* #533 **Batch Payments for select Call Centre**
* #489 **September Billing**
* #572 **October billing files**
* #573 **November billing Files**

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Sep 13, 2017, 24:16

from Kyle
@smasih @tim.traylor

Thanks, I performed the following steps:

*     I reset the Billing cycle date to 7/26/17 from the site edit page
*     I deleted the current billing cycle
*     I refreshed the page to have it reflect then the 'next' billing cycle date of 8/25/17
*     I clicked 'create' to create the billing cycle.
*     Was presented with "something went wrong error"

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Sep 13, 2017, 24:17

We have resolved the error please try again and I am creating a new ticket #468 as this one holds so many comments due to which browser get's stucked for a while.

So from now onwards can we please post comments on #468

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 13, 2017, 08:43

@smasih @tim.traylor 

I performed the following steps:

* Reset the Billing cycle date to 7/26/17
* Went to the billing cycle page and saw correct cycle date of 8/23/17
* Clicked "create"
* Selected file uploaded and then clicked upload
* Was presented with "There is no cdr mapping directory for this site"


@smasih 

Judging from the back to back errors, may the Aavaz team please complete the upload and finalization of the billing cycle; allowing for the ability fix any other issues that are incurred. Thank you

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Sep 15, 2017, 08:20

@kdjstudios 

I have finalized the billing cycle, but the total seems to be off, many accounts are not included in **Aug_23_inv_detail.xlsx** but were included in usage cycle.

can we get [dkc7gH.pdf](/uploads/eaab26df8c16ffd87a55b44960e9b769/dkc7gH.pdf) for August.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 15, 2017, 09:01

@smasih 

The PDF file was already uploaded earlier, but we now have it in spreadsheet for easier comparisons. 

Also the totals are completely different.  Their Billing = 369_105.44; while our Billing = 275,133.14

 [Aug_23_inv_detail__1_.xlsx](/uploads/c360c54cdceff227c5f9346908a00811/Aug_23_inv_detail__1_.xlsx)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 19, 2017, 10:10

[explanation_of_billing_data_04-20-2017.docx](/uploads/2c9cb6e8beae1ee6e70a620e681df00d/explanation_of_billing_data_04-20-2017.docx)

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Sep 20, 2017, 07:22

@kdjstudios   
Before we move forward, can you please confirm from **Garrett** that our findings are correct.   
There are 2 handle # assign to each account.  

1.  for the activities/ monthly transactions
1.  for the recurring charges

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 20, 2017, 08:39

@pchaudhary 

We will not need to bother Gary with the technical side of the conversion. If you have any questions please relate them to @tim.traylor and I. And as confirmed yesterday along with the orginal import there can be multiple handles per account number.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Sep 20, 2017, 09:08

@kdjstudios 

Sorry, there was a typo. It was Garrett not Gary (I have edited the comment)

**And as confirmed yesterday along with the  original import there can be multiple handles per account number**

Thanks for confirming from your end Kyle but at the moment we do not have support to serve two handles, so I just wanted to be certain that we are moving in right direction.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 20, 2017, 10:16

@tim.traylor @smasih @pchaudhary 

This was all discussed yesterday, why was the orginal import able to import multiple handles to the same account, or was that done incorrectly as @tim.traylor pointed out?

The import files have **always** contained multiple handles. Please **correct** your import process to handle this. Also as @tim.traylor pointed out yesterday, we should not be finding out about issues like this 5 months into the conversion process.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Sep 26, 2017, 12:29

@tim.traylor 

I was trying to re run the import and billing process, but there is something I am not able to understand.

We came to know that there are two handles, one for recurring and one for transactions (activities) which is fine, we don't need to build a support for that because, recurring fee is automatically calculated during invoice generation.

BUT, after not getting the desired result we found we are using both handles for recurring and transactions.

(*rcr1.csv*)  

| handle   | charge_type | frequency | account | account_dept | service_type | ever | anniversary | price | telephone    | quantity | first_period | origin | salesperson |
|----------|-------------|-----------|---------|--------------|--------------|------|-------------|-------|--------------|----------|--------------|--------|-------------|
| TAG-6708 | TAS         | M         | 102384  |              | TAS          | A    | 1           | 295   | 780-538-6708 | 1        | 241          |        | 17-09-19    |
| TAG-6711 | TAS         | M         | 102384  |              | TAS          | A    | 1           | 400   | 780-538-6711 | 1        | 254          |        | 17-09-19    |
| TAG-6708 | ZRCD        | M         | 102384  |              | TAS          | A    | 1           | 0     | 780-538-6708 | 1        | 197          |        | 17-09-19    |
| TAG-6708 | ZFLT        | M         | 102384  |              | TAS          | A    | 1           | 0     | 780-538-6708 | 1        | 197          |        | 17-09-19    |
| TAG-6708 | ZFAX        | M         | 102384  |              | TAS          | A    | 1           | 0     | 780-538-6708 | 1        | 197          |        | 17-09-19    |
| TAG-6708 | ZEML        | M         | 102384  |              | TAS          | A    | 1           | 0     | 780-538-6708 | 1        | 197          |        | 17-09-19    |
| TAG-6711 | ZRCD        | M         | 102384  |              | TAS          | A    | 1           | 0     | 780-538-6711 | 1        | 197          |        | 17-09-19    |
| TAG-6711 | ZFLT        | M         | 102384  |              | TAS          | A    | 1           | 0     | 780-538-6711 | 1        | 197          |        | 17-09-19    |
| TAG-6711 | ZFAX        | M         | 102384  |              | TAS          | A    | 1           | 0     | 780-538-6711 | 1        | 197          |        | 17-09-19    |
| TAG-6711 | ZEML        | M         | 102384  |              | TAS          | A    | 1           | 0     | 780-538-6711 | 1        | 197          |        | 17-09-19    |

|

As in the above table you can see the account is same but two handles with different values for the same service.

Now let's pull the **Aug_23_inv_detail_1.xlsx** which contains final calculations. In this one only one of the service is billed.

**So, our question is.. which one to import ..??**

| Invoice | Account | Account Name              | Freq Code   | Service | Charge | Qty | Price |
|---------|---------|---------------------------|-------------|---------|--------|-----|-------|
| 5125976 | 102384  | Husky Oil Operations Ltd. | Recurring M | TAS     | ZFAX   | 1   | 0     |
|         |         |                           |             | TAS     | ZFLT   | 1   | 0     |
|         |         |                           |             | TAS     | ZRCD   | 1   | 0     |
|         |         |                           |             | TAS     | TAS    | 1   | 400   |
|         |         |                           |             | TAS     | ZEML   | 1   | 0     |
|         |         |                           |             | TAS     | ZFAX   | 1   | 0     |
|         |         |                           |             | TAS     | ZFLT   | 1   | 0     |
|         |         |                           |             | TAS     | ZRCD   | 1   | 0     |


**This one is Invoice Detail Report of september billing #489** Which makes clear that all of it needs to be imported, even if the recurring fee is repeated with different value, **but when to bill the particular recurring fee (both had frequency "M" which means we need to bill both of them monthly )**

| Invoice | Account | Account Name              | Freq Code   | Service | Charge | Qty | Price |
|---------|---------|---------------------------|-------------|---------|--------|-----|-------|
| 5125976 | 102384  | Husky Oil Operations Ltd. | Recurring M | TAS     | TAS    | 1   | 295   |
|         |         |                           |             | TAS     | ZEML   | 1   | 0     |
|         |         |                           |             | TAS     | ZFAX   | 1   | 0     |
|         |         |                           |             | TAS     | ZFLT   | 1   | 0     |
|         |         |                           |             | TAS     | ZRCD   | 1   | 0     |
|         |         |                           |             | TAS     | TAS    | 1   | 400   |
|         |         |                           |             | TAS     | ZEML   | 1   | 0     |
|         |         |                           |             | TAS     | ZFAX   | 1   | 0     |
|         |         |                           |             | TAS     | ZFLT   | 1   | 0     |
|         |         |                           |             | TAS     | ZRCD   | 1   | 0     |

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 26, 2017, 13:30

@smasih @tim.traylor 

I would say that every one of the 'handles' needs to be imported and assigned. Then the client (Garrett) can choose which one (or both) to keep active or deactivate accordingly, if that leads to discrepancy in the billing invoices.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Sep 28, 2017, 07:33

@smasih @kdjstudios Hi Guys, So....  Here is my understanding of what I see in the files and what I think you are telling me:

1. Handles are tied to a package, which describes how/what get billed.
1. There can be multiple handles per-account. This means that they can have multiple ways of billing the same thing (e.g., they could have two different prices for talk time for the same account, and which we use depends on the handle).  This was stated by the client in the explanation document they provided.
1. In the usage files the handles are the identifier of the records.
1. In building the accounts you had been using the handle as the resource ID.  However, since there can be multiple handles per-account, this will not work.

Assuming what I described is correct, I think that the only way to handle it is with parent/child accounts.  That would allow for different billing packages, with the handle being used as the resource ID so it matches the usage import data.

Let's review this at the 11:00 meeting (I think Gary will be missing it again because he is travelling) and then review it with the client if we're in agreement this will work.

thx,
Tim

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 28, 2017, 09:01

Tim I do not think we should be using the handle # on the accounts at all. This was to my understanding to be used just during the import process to associate packages (activities) with accounts, and not to be saved anywhere to the account. I think the Resource ID should match the account number.

![Untitled](/uploads/6c26de7347e1b76c8430ca0a0337a00b/Untitled.png)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 28, 2017, 09:20

Can we confirm that in the multiple handles that there is only once that is usage rate and the other handles are a base rate?

If there are multiple usage rates with different rates of charge and packages please provide an example. - This should be a form of tiered pricing.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 28, 2017, 15:29

@smasih @tim.traylor 

After manually looking though Tas2 file and highlighting in RED duplicate accounts to narrow down to accounts with multiple handles, then I highlighted of those accounts in YELLOW the ones that have different Packages, there are only THREE accounts total that have multiple rates packages specified. ALL the rest were either duplicate copies or multiple flat rate charges. 

Once imported correctly, I will reach out to Garrett to decide which charges on those three accounts need to be deactivated or if those three accounts need subaccounts. 

[SEP20_17_b2tas2_KDJ_Highlighted.xlsx](/uploads/438d18ef5ee3c49a83a34879464aae01/SEP20_17_b2tas2_KDJ_Highlighted.xlsx)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 28, 2017, 15:32

@tim.traylor @smasih 

We should be using the Account number as both the Account Number and the Resource ID in SA Billing. 

Judging from the invoice exports, they do not use the Handle to determine accounts. That appears to be more or less a unique ID or index that was assigned to each 'activity' it appears.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Sep 29, 2017, 03:52

@kdjstudios
* **We should be using the Account number as both the Account Number and the Resource ID in SA Billing. **
 * Yes we can do that which would be ideal but in usage file the usage values are printed against the handle number.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Sep 29, 2017, 03:56

@tim.traylor @kdjstudios 

We can have activities and additional fees imported for those accounts who just have one handle, for the rest I will test out those with child accounts. 

So, if Garrett agrees or this is the case we would save some development time Instead of waiting for him to get back to us.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Sep 29, 2017, 06:27

@kdjstudios Hi Kyle, I understand you identified three accounts that have multiple rate packages. Your comment seems to assume that some should be deactivated, but I don't see any comment indicating that you checked with Garett and that he said the multiple packages are invalid or incorrect data.  Please clarify.  Thx, Tim

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 29, 2017, 08:17

@smasih 

Thanks summit, agreed lets go ahead with @tim.traylor sugesstion from yesterday to import as child accounts.

@tim.traylor 

I have not checked anything with Garrett, as I would like to have the data in SA Billing, so that I can point out the accounts to him and confirm the setup in our system and compare to how it is setup in his system. I will send him an email today and CC you on it, with a further explanation into this issue.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 29, 2017, 11:13

@smasih @tim.traylor 

**Notes from call with Garrett:**


*  Each handle will have a separate child account for those with duplicates but they will child accounts bill into a parent account; not separately. We may have to manually specify the parent account from the child accounts. As setting up the parent accounts we were unable to see a notation of that in the tas files.


*  The CMR file will only have the accounts with balances too, so some of these handles may not match any accounts in the CMR we should not need to import those as those are just old(removed) accounts.


*  The z-internal is just used for internal tracking and there no invoice or rates associated with those charges.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 29, 2017, 11:29

I also want to confirm @smasih you were correct that we need to have the Resource ID be the handle.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Oct 5, 2017, 07:17

* Account: "104657 - Lyons Production Services Ltd."
* Account: "104528 - Phoenix Technology Services LP"  
Both of the accounts have recurring fees.  

**rcr1.csv**

| handle   | charge_type | frequency | account | account_dept | service_type | ever | unused1 | unused2 | start_date | end_date   | date_suspend | anniversary | price | telephone    | quantity | first_period |
|----------|-------------|-----------|---------|--------------|--------------|------|---------|---------|------------|------------|--------------|-------------|-------|--------------|----------|--------------|
| TAG-6434 | ZEML        | M         | 104657  |              | TAS          | A    |         |         | 2003-07-02 | 0000-00-00 | 0000-00-00   | 1           | 0     | 780-814-6434 | 1        | 200          |
| TAG-6434 | ZFAX        | M         | 104657  |              | TAS          | A    |         |         | 2003-07-02 | 0000-00-00 | 0000-00-00   | 1           | 0     | 780-814-6434 | 1        | 200          |
| TAG-6434 | ZFLT        | M         | 104657  |              | TAS          | A    |         |         | 2003-07-02 | 0000-00-00 | 0000-00-00   | 1           | 0     | 780-814-6434 | 1        | 200          |
| TAG-6434 | TAS         | M         | 104657  |              | TAS          | A    |         |         | 2010-05-12 | 0000-00-00 | 0000-00-00   | 1           | 347   | 780-814-6434 | 1        | 289          |
| TAG-6434 | ZRCD        | M         | 104657  |              | TAS          | A    |         |         | 2003-07-02 | 0000-00-00 | 0000-00-00   | 1           | 0     | 780-814-6434 | 1        | 200          |
| TAC-0695 | ZRCD | M | 104528 |   | TAS | A |   |   | 2003-04-23 | 0000-00-00 | 0000-00-00 | 1 | 0   | 403-212-0695 | 1 | 197 |
| TAC-0695 | TAS  | M | 104528 |   | TAS | A |   |   | 2015-07-29 | 0000-00-00 | 0000-00-00 | 1 | 755 | 403-212-0695 | 1 | 357 |
| TAC-0695 | ZEML | M | 104528 |   | TAS | A |   |   | 2003-04-23 | 0000-00-00 | 0000-00-00 | 1 | 0   | 403-212-0695 | 1 | 197 |
| TAC-0695 | ZFLT | M | 104528 |   | TAS | A |   |   | 2003-04-23 | 0000-00-00 | 0000-00-00 | 1 | 0   | 403-212-0695 | 1 | 197 |
| TAC-0695 | ZFAX | M | 104528 |   | TAS | A |   |   | 2003-04-23 | 0000-00-00 | 0000-00-00 | 1 | 0   | 403-212-0695 | 1 | 197 |

But these are not getting billed or say these accounts are not available in [Invoice Details sheet ](http://gitlab.aavaz.biz/AnswerNet/SABilling/uploads/c360c54cdceff227c5f9346908a00811/Aug_23_inv_detail__1_.xlsx) provided by Garrett however, these accounts were present in usage csv file.  

**The second Scenario is interest calculation, I am not sure how they are calculating interest**

Accounts like 
* 104865 - Surepoint Technolgies Group  
* 104864 - Cardinal Condominiums

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 5, 2017, 09:02

@smasih @tim.traylor 

Well I think we are getting closer to being back on track. If I am looking at the two billing totals correctly today; then the Invoice detail sheets total is $369105.44; while our billing cycle detail report shows a total of $359,525.09. This is only a discrepancy of $9580.35 or less then 3%. 

Can we put together a list of accounts that do not match; I can then reach out to Garrett for clarification on all of these accounts?

Also I noticed we are using the wrong billing cycle date in SAB. It appears to be under Feburary, while this cycle was for August.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Oct 10, 2017, 09:19

Hi, [Kyle](@kdjstudios) did we manage to get any reply from Garrett..?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 10, 2017, 09:33

@smasih 

There was no response from Garrett over the weekend, I will reach out again today for an update.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 12, 2017, 14:53

@smasih @tim.traylor 

I received a Phone call from Garrett today and below is the information he was able to update me with. I have also requested for him to send an email in his own words to insure I am correctly explaining everything. I will attach that once I have received it. 

1.  ZINT Charge is the late fee charge. The late fee charge is 2% and is only charged on accounts over 56 days (2 billing cycles)

2.  Tax rates:  Exempt 0%; AB (Alberta) 5%; ON (Ontario) 13%

3.  With the missing accounts they found that those accounts are located on their PDF document but not on their Excel document. I have requested for them two start sending both editions, so we can triple check the numbers with both files to insure accuracy.

[Aug_23_inv_detail.dat](/uploads/b3fbaf78e3dbc7d9534f71be2f6cd331/Aug_23_inv_detail.dat)

[Aug_23_inv_detail-1.xlsx](/uploads/ddd02304414edbe80be2996619d3d45e/Aug_23_inv_detail-1.xlsx)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 16, 2017, 13:45

@smasih @tim.traylor 

As you may have noticed there are quite a few accounts missing from their excel export that are listed on the dat/pdf file. I have converted the dat file to pdf and to excel using the https://www.pdftoexcel.com/ and believe it has done a successful job at insuring all data is listed. 

[Aug_23_inv_detail__2_.xlsx](/uploads/729e42da9584b0591029a255adebd983/Aug_23_inv_detail__2_.xlsx)

[Aug_23_inv_detail.pdf](/uploads/aef1d29e19f30fd840fb2cbf2ac4f3eb/Aug_23_inv_detail.pdf)

Please let us know if there is anything else we can do, we need to move forward on this urgently.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Oct 17, 2017, 07:55

@kdjstudios   
first of all thanks a lot for pdf and xlsx.

* ZINT Charge is the late fee charge. The late fee charge is 2% and is only charged on accounts over 56 days (2 billing cycles)
  * Is there a minimum value for late fee to be charged (eg: Late fee should not be charged for amount below $5)
  * Is there a minimum amount for late fee..? (eg: Late fee should not be less than some $x amount in any scenario)
  * Is late fee taxable..?
  * **How to decide that the account is unpaid for 56 days as we have no previous invoice..!! @tim.traylor  may we please think of a workaround as we do have info on aging buckets from Customer Import Sheet (but we haven't imported that info yet)**

* Tax rates: Exempt 0%; AB (Alberta) 5%; ON (Ontario) 13%
 * From Garrett  `` Also he did want us to pay attention to and check the 'interest' and 'tax' charges on accounts. Note the tax column 'K' - ' tax_district' there should be two tax rates, one for Alberta at 5% and another for Ontario at a different rate (he didn't have this rate on hand but said will get back to me accordingly). If marked Exempt there is not Tax (Government accounts). The interest is specified in column 'S' - 'supp_int' and is defined as "A" is Yes it is charged and "N" is No it is not charged.`` 
 * Now we have calculated Tax on the most of the accounts although the calculation is correct but its missing form the **PDF given to us** 
 * For eg here is an account with **5%** of tax

| Account Number | Account Name | Invoice            | Line Item Description | Line Item Amount | Tax Code | Tax Amount |
|----------------|--------------|--------------------|-----------------------|------------------|----------|------------|
| 102351         | TAG-2026     | Arc Resources Ltd. | SCC-140               |                  |          |            |
|                |              |                    | C-NALP/ CALL          | $0               |          |            |
|                |              |                    | C-NALP/ CALL          | $0               |          |            |
|                |              |                    | TAS                   | $100             |          |            |
|                |              |                    | ZEML                  | $0               |          |            |
|                |              |                    | ALRM                  | $35              |          |            |
|                |              |                    | TAS                   | $345             |          |            |
|                |              |                    | ZRCD                  | $0               |          |            |
|                |              |                    | ZFLT                  | $0               |          |            |
|                |              |                    | ZFAX                  | $0               |          |            |
|                |              |                    |                       |                  | AB       | $24        |

* we got this from import file

| account | name               | address1            | address2           | address3 | address4 | address5 | zip     | contact | telephone    | tax_district | sic | po | invoice_detail | sort | flip | advance | supp_prev | supp_int | supp_coll | supp_auto | supp_beeps | supp_print | term_reason | coll_agency | company | salesperson | account_type | account_suspend | co_inv_sort | po_expiry | inv_copies | inv_cycles | inv_anniversary | start_date | term_date  | date_last_inv | date_last_payment | tier_over | total_balance | 30_balance | 60_balance | 90_balance | 120_balance | instant_inv_debits | amt_lat_payment | security_depost | year_to_date_interest | credit_limit | due_not_inv | deposit_not_inv | undo_instant_inv | total_credits_since_last_inv | total_debits_since_last_inv | last_reference | last_invoice | inv_period_last_modified | number_notes | number_perm_notes | pop_up_id | credit_card_code | auto_debit | fax          | credit_card_no | credit_card_name  | inv_adv_last_peroid | sales_ref | undo_balance | 30_undo_balance | 60_undo_balance | 90_undo_balance | 120_undo_balance | undo_deposits | undo_interest | locked | instant_inv_this_period | main_or_sub | cycle | alt_lang |
|---------|--------------------|---------------------|--------------------|----------|----------|----------|---------|---------|--------------|--------------|-----|----|----------------|------|------|---------|-----------|----------|-----------|-----------|------------|------------|-------------|-------------|---------|-------------|--------------|-----------------|-------------|-----------|------------|------------|-----------------|------------|------------|---------------|-------------------|-----------|---------------|------------|------------|------------|-------------|--------------------|-----------------|-----------------|-----------------------|--------------|-------------|-----------------|------------------|------------------------------|-----------------------------|----------------|--------------|--------------------------|--------------|-------------------|-----------|------------------|------------|--------------|----------------|-------------------|---------------------|-----------|--------------|-----------------|-----------------|-----------------|------------------|---------------|---------------|--------|-------------------------|-------------|-------|----------|
| 102351  | Arc Resources Ltd. | 1040 - 9909 102 St. | Grande Prairie, AB |          |          |          | T8V 2V4 |         | 403-503-8600 | AB           |     |    | T              | O    | N    | 1       | N         | A        | N         | N         | N          | N          |             |             | 1G      | GRACE       | B            |                 | 1G          |           | 1          | 1          | 6               | 2000-03-27 | 0000-00-00 | 2017-08-23    | 2017-08-10        | 0         | 504           | 0          | 0          | 0          | 0           | 0                  | -504            | 0               | 229.63                | 0            | 0           | 0               | 0                | 0                            | 0                           | 5124898        | 5126082      | 384                      | 2            | 2                 | 0         |                  | N          | 403-503-8733 |                | Arc Resouces Ltd. | 1                   | GP        | 0            | 0               | 0               | 0               | 0                | 0             | 229.63        |        | N                       |             | 385   | N        |

* **So, are we doing something wrong or is there a different report for "Taxes" other than this "PDF file"..?**   

|

* With the missing accounts they found that those accounts are located on their PDF document but not on their Excel document. 
  * This was a good news, our system was importing and processing data perfectly.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 17, 2017, 08:06

@smasih 

I have just reached out to Garrett with these questions:
* Is there a minimum value for late fee to be charged (eg: Late fee should not be charged for amount below $5)
* Is there a minimum amount for late fee..? (eg: Late fee should not be less than some $x amount in any scenario)
* Is late fee taxable..?

In regards to checking the Taxes, I have reached out to him to see if there is another report for that. For now though, we should be checking the line items of the invoice to insure those are correct.

Lastly, yes that is great that our data is matching the PDF export. I expect the converted PDF to Excel file I uploaded contains all of the needed data, however if you have a different prefered method of converting PDF to Excel please feel free to use it.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 17, 2017, 09:53

@smasih @tim.traylor 

I quickly filtered and totaled the ZINT charges from the Aug 23 In detail (2) file above. Please double check that you are able to filter and sort those same charges and totals. Roughly 1100 in late fees is the discrepancy we should be aiming for. 

[ZINT_Charges__Late_Fees__Aug.xlsx](/uploads/f9adfdea078c0a800f146e746e8b69e4/ZINT_Charges__Late_Fees__Aug.xlsx)

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Oct 18, 2017, 05:35

@kdjstudios 

* Thanks for consolidating ZINT charges, now we know what to add on our billing cycle total.

* We need an update on TAX calculation or if it's a different report.

* We need to know what are **ACT** and **DIS** charges ..?? As these were not present in any of the import files.

* As per SAB code we  **do not** charge terminated accounts for recurring charges. But there are few accounts in the **Invoice Detail Report from Garrett** where terminated accounts has been charged for the recurring fees. for eg:
   * 105838 Anderson, Betty
   * 106576 Fuse Realty  

  * **Question here is.. do we need to make changes in our system to support their functionality or there is something else we are missing on.**

* Lastly, we are **$8213.54** behind from their total of **$369,105.44** (_after adding ZINT charges $1084.91 to our total $359,806.99 excluding Tax charges_)

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Oct 18, 2017, 05:37

Just FYI if  anyone wants to go ahead and research for faulty accounts they can do it by pulling "Billing Cycle Detail Report" on **dev.sabilling.com** and compare it with the [Aug_23_inv_detail__2_.xlsx](http://gitlab.aavaz.biz/AnswerNet/SABilling/uploads/729e42da9584b0591029a255adebd983/Aug_23_inv_detail__2_.xlsx)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 18, 2017, 08:10

@smasih @tim.traylor 

* When I receive a response from Garrett I will update the team on his report for Tax Charges, but currently as they are not listed in the Invoice Detail report, lets disregard the tax charges and continue to compare the line item charges. 

* ACT and DIS charges appear to be 'One Time' Charges according to the majority, and I would presume mean: Activation and Disconnect. These are marked also as TAS charges and we should be able to ignore them or manually take into account the SUM of the ACT and DIS charges are only $186.25.

* I will also reach out to Garrett for an answer about charging terminated accounts recurring fees. if there are only a few accounts we should be manually totaling those out and comparing the discrepancy, if any. 

* Lastly, in regards to researching and checking accounts, I thought that is what the Aavaz team was doing the last few weeks? Has this not been accomplished?

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Oct 25, 2017, 09:16

@kdjstudios @kace.coady @tim.traylor 

In addition to previous queries I have another one for Garrett 

* **What is _Prorated A_ and _Prorated M_ under _Freq Code_ column in [Aug_23_inv_detail__2_.xlsx](http://gitlab.aavaz.biz/AnswerNet/SABilling/uploads/729e42da9584b0591029a255adebd983/Aug_23_inv_detail__2_.xlsx) as they have not been defined in any of the import files.**  

* Total of *Aug_23_inv_detail__2_.xlsx* does not seems accurate to me.

* Still assuming the total is correct and excluding undefined charges(*ZINT, ACT,DIS, Prorated A/M*) and taxes we are now **$5371.92** behind from their total of $369,105.44

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 25, 2017, 09:35

@smasih @tim.traylor @kace.coady @abakshi 

Hello Everyone, 

I have put together an excel spreadsheet of all the account balances from both SCC and SAB and then ran a formula to check for differences in the same account totals. Note that the rows do not have to line up with the same account number, I compare throughout the whole spreadsheet. These are displayed in column G "SCC Match Charge" if there was no account in SCC that matched an account number in SAB then "#N/A" is the result. We should first be looking into why these accounts are in SAB but not in SCC. 

I then also calculated the difference in account totals per each account which is displayed in Column H "Difference". I highlighted all cells that were not equal to zero, please note some values are off by factions of a cent due to excel's rounding and please ignore them. Example row 28.

[SCC_vs_SAB_Invoice_total_Comparisons.xlsx](/uploads/98eed5c93c578c2a7261f5c4104398d6/SCC_vs_SAB_Invoice_total_Comparisons.xlsx)

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Oct 26, 2017, 01:12

 @kace.coady  @abakshi @tim.traylor 

@kdjstudios Thanks for the sheet It was really helpful. I got **11** accounts which had a different Quarter for billing **rest of them were just $0 as total**

Although it wasn't specified in any of the import but was pretty evident after going through the report.

I have ran the billing cycle process again please have look again.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 26, 2017, 08:52

@smasih @kace.coady @tim.traylor 

I do not know where you only got **11** accounts from. I have filtered the document to only show accounts with differences greater then $0.01 and I am showing **147** accounts.

[Accounts_with_Discrepency.xlsx](/uploads/a294fafbbbe8609c20aa345ac393329b/Accounts_with_Discrepency.xlsx)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 26, 2017, 09:47

During the conference call it was mentioned that Atul and Sumeet will be sending and email detailing out my comparisons and further explaining what they are needing.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 26, 2017, 10:41

I have rerun the spreadsheet comparisons on the new billing cycle. we are still at a difference of about $8000

Only the accounts (About 140) with differences greater then $0.01:
[Accounts_with_Discrepency_10-26-17.xlsx](/uploads/9c2822cc97ac82a40acf7ea6b24c17d0/Accounts_with_Discrepency_10-26-17.xlsx) 

All accounts comparison:
[SCC_vs_SAB_Invoice_total_Comparisons_10-26-17.xlsx](/uploads/1d23b7b2fd4c819b73c6d972cc023328/SCC_vs_SAB_Invoice_total_Comparisons_10-26-17.xlsx)

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Oct 30, 2017, 03:22

@kdjstudios 

I just ran the SCC invoice detail report against our system and results are in this file[adiff.csv](/uploads/3778cf15a2846212c459406490f1720b/adiff.csv) using a script.  `I have also  e-mailed the file to you, incase the uploaded file link is not working.`

Following were the major differences
*  Two charges **Prorated A** & **Prorated M** as both of them were not defined in the import files or maybe I missed something. 
* ZINT (Zint is the late fee charges so we will be putting that aside)
*  ACT, DIS Codes (These codes were also not defined in the import perhaps these were manually added charges)
* Invoice of deactivated accounts. (SAB does not bill inactive/terminated account for recurring or usage charges **but these accounts were billed on SCC report**. I might have missed on anything but It would be great if we can get an update from Garrett.) 
* There were few accounts which were not consumed properly by our import script so I have manually fixed those accounts.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Oct 30, 2017, 03:50

moving to next milestone

**Comment by kdjstudios** [comment]
In GitLab by @kace.coady on Oct 30, 2017, 05:58

Kyle- Please review

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 30, 2017, 10:56

@smasih @kace.coady @tim.traylor 

Thank you for the update. 

* Do we need more information from Garrett about the Prorated A and Prorated M fees? - I would presume that they would be charged to new/terminated accounts that are receiving a prorated rates for partial activity during a billing cycle rather then the full billing cycle charges?
* What is the total of the ZINT charges again? So we can insure not to take that amount out of the difference total.
* Do we need more information from Garrett about the ACT and DIS fees? - I believe it was previously mentioned that the total of these fees was under $200 correct and shouldn't effect much of the difference. 
* I thought only deactivated terminated accounts would not bill if the account balance (or was it invoice balance) was equal to zero? I will reach out to Garrett in regards to those accounts that were marked terminated but were billed.
* In regards to fixing the incorrect accounts, will this fix maintain on the next import? Or will these need to be manually fixed again after each import? 

I ran the Billing cycle detail report ([billing_cycle_detail_10-23-2017_Master.pdf](/uploads/1617fc158ff8daea7f7f4ffb3c57fd9e/billing_cycle_detail_10-23-2017_Master.pdf)) and we have a total billing of: 362194.44

SCC export (AUG 23 Invoice Detail) shows a total billing of: 369105.44

This shows a difference of: 6911

But the total difference outlined in adiff is: 6304.43

Am I missing some information or how is your reported difference not equal to the difference I just calculated. I know it is only roughly 400, but just a concern I noticed.


*Also in overview with about a 7000 difference we are now down to roughly a 2% discrepancy from the SCC total. With approval from @kace.coady or @tim.traylor ; I think this would be acceptable, now that it appears we know the main causes for these issues and can deduce that about 100 accounts are all that is affected; that we proceed to the Septembers exports and billing files and begin checking those. Then if all checks there too, we can do the October billing cycle. At that point I hope we will be inline to do the November cycle in a side by side, make any needed manual changes and then release them to production for Dec?!*

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 31, 2017, 10:02

I will reach out to Garrett to confirm the Prorated Fees and confirm how they bill their terminated accounts ( charged base rate and usage).  I will also start discussion for the Oct. billing cycle files and exports. 

The main discrepancy is that SCC is billing their terminated accounts. 

@kace.coady is approving moving to the September cycle. @smasih will be creating a new script to new import data, rather then completely reimporting. Will be primarily be focusing on the account invoice balances, rather then the account balances.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 31, 2017, 16:18

Update from Previous questions from Garrett:

1.  Is there a minimum value for a late fee to be charged? (ex: Late fee should not be charged for amount below $5)
2.  Is there a minimum amount for late fee? (ex: Late fee should not be less than some $X amount in any scenario)
3.  Is late fee taxable?
4.  Do you have an export or report for the last invoices that includes or shows the tax? (This is so we can check our tax calculations on the accounts)


> 1. minimum value for a late fee. no there is not
> 2. minimum amount for late fee, yes, 5 dollars. 
> 3. taxable, no 
> 4. I've attached a report converted to excel, with is the last billing period, the 18 of oct, which has each invoice, account name, recurring charge amount and the tax to each invoice. 

Let me know if you have questions. thanks. 

[TaxReportOct-18-2017-1.xlsx](/uploads/8c53268ef0a1891ca085a63b01dd2b2d/TaxReportOct-18-2017-1.xlsx)

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Nov 7, 2017, 24:28

@kdjstudios @tim.traylor 

Seems like tax calculation is fine as well but, there is a little discrepancy and that too because of difference in invoice totals.

our tax total: **$17,913.88**  
scc total: **$17530.91**
diff: **$382.97**

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 7, 2017, 13:20

@smasih 

This looks promising!

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jan 9, 2018, 05:04

Moving to current milestone.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jan 9, 2018, 05:04

removed milestone

**Comment by kdjstudios** [comment]
In GitLab by @abakshi on May 8, 2018, 06:05

removed milestone

**Comment by kdjstudios** [comment]
In GitLab by @abakshi on May 8, 2018, 06:05

@kdjstudios @tim.traylor 

Can we close this ticket?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 8, 2018, 11:42

Yea might as well.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 30, 2018, 08:21

closed
